src/transformers/modeling_utils.py

```diff
@@ -2368,13 +2370,39 @@ def from_pretrained(
                 " ignored."
             )
 
+        if commit_hash is None:
```

Comment:
If I understand correctly, we can have 2 different origins for the commit hash:
- kwargs
- saved in the config file
and we were only checking the `kwargs` when looking for the `maybe_adapter_model_path ` which means that  specific revisions were not loaded correctly.
Won't this early call not solve the issue for local files ( when `is_offline_mode()`) ? 

What I mean here is that we are looking for the file online twice, here and after with ` cls.config_class.from_pretrained(` that has the correct `local_files_only`. and other kwargs set 
Can we move this logic above / use the `resolved_config_file`.

Reply:
No, the commit hash is not saved in the config file. It is recovered from the call to get the config. And commit_hash is not relevant in offline mode, since then we don't make any calls to the Hub at all.